lib/reek/smells/unneeded_suppression.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Reek
+  module Smells
+    #
+    # An unneeded suppression is a Reek's comment disabling a unexistent smell.
+    #
+    # See {file:docs/Unneeded-Suppression.md} for details.
+    class UnneededSuppression
+      def initialize(context, smell_detector)
+        @context = context
+        @smell_detector = smell_detector
+      end
+
+      def sniff
+        if @smell_detector.sniff(@context).empty?
+          exp = @context.exp
+          ctx_source = exp.loc.expression.source_buffer.name
+          [SmellWarning.new(self,
```

Comment:
I see. But then maybe the overall architecture is not the right one.
I mean, we **need** stuff that is in SmellDetector (otherwise we're just duplicating code 1:1 like we do here) but at the same time we try really hard to **not** make this a smell detector.
That doesn't seem right.
How about not making this a detector-style thingie but instead follow the same approach you, @mvz, suggested in #1082 at the end?
So:

>> I have in mind to handle it at the Examiner level. So here we would first catch Reek::BadConfigurationError, and then StandardError.

wdyt?